packages/cadl-ranch-specs/http/special-headers/repeatability/mockapi.ts

```diff
@@ -0,0 +1,25 @@
+import { passOnSuccess, ScenarioMockApi, mockapi, ValidationError } from "@azure-tools/cadl-ranch-api";
+
+export const Scenarios: Record<string, ScenarioMockApi> = {};
+
+Scenarios.SpecialHeaders_Repeatability_doSomething = passOnSuccess(
+  mockapi.post("/special-headers/repeatability/doSomething", (req) => {
+    if (!("repeatability-request-id" in req.headers)) {
+      throw new ValidationError(
+        "Repeatability-Request-ID is missing",
+        "A UUID string",
+        req.headers["Repeatability-Request-ID"],
+      );
+    }
+    if (!("repeatability-first-sent" in req.headers)) {
+      throw new ValidationError(
+        "Repeatability-First-Sent is missing",
+        "A date-time in headers format",
+        req.headers["Repeatability-First-Sent"],
+      );
+    }
+    return {
+      status: 204,
+    };
```

Comment:
Ideally, the test should return 5xx upon first request (simulate service error), and return 204 upon second request (the retry), and verify that value of both header is exactly same on the first request and the retry.

Though how best to do this in mockapi is a bit beyond me.

Reply:
I believe testing that the retry behavior of azure-core is outside of the scope of this test. The scope of this test is to make sure the codegen do not generate parameters, and send random values and "now".

Reply:
Got it. Java will then keep its local mock test for verifying the correctness of retry behavior.

Reply:
@lmazuel I can update this test to include all the other scenarios. Just worried if we just include this one like htat it basically will come down to generators having to reupdate it after

Reply:
I think we can start with this one. My major concern is the name. It could be `doSomething` or `post` (if we plan adding `put` and `delete`).

Reply:
yeah that's true, I'm leaning more towards `immediateSuccess` because that's what shows up in the dashboard and will make more sense as to what it is doing.